core/domain/exp_services.py

```diff
@@ -1961,13 +1922,13 @@ def create_or_update_draft(
             exp_user_data.draft_change_list_last_updated > current_datetime):
         return
 
-    updated_exploration = apply_change_list(exp_id, change_list)
-    updated_exploration.validate(strict=False)
-
     if exp_user_data is None:
         exp_user_data = user_models.ExplorationUserDataModel.create(
             user_id, exp_id)
 
+    updated_exploration = apply_change_list(exp_id, change_list)
+    updated_exploration.validate(strict=False)
```

Comment:
I think firstly we should make sure that the change_list is correct then only create a ExplorationUserDataModel. Is there any other reason we are changing this?

Reply:
You see if we are to get an empty ExplorationUserDataModel the apply_change_list function would throw error and line 1925 would never get called. Hence to make line 1925 callable, I readjusted the blocks of code

Reply:
Can you explain `if we are to get an empty ExplorationUserDataModel the apply_change_list function would throw error `?
I don't think `apply_change_list` method depends on ExplorationUserDataModel.

Reply:
apply_change_list checks for get_exploration_by_id() and that will throw error if there is an invalid exp_id. When we create an exploration, the ExplorationUserDataModel also gets created right?

Reply:
> When we create an exploration, the ExplorationUserDataModel also gets created right?

I don't think so can you please check?

> apply_change_list checks for get_exploration_by_id() and that will throw error if there is an invalid exp_id.

Yeah, that's why I'm saying if we know that `apply_change_list` can through an error if `exploration doesn't exist` or `change_list is incorrect` then let's create the new `ExplorationUserDataModel` if it passes the `apply_change_list` method and exploration validation checks.

Reply:
Yes I have tried --The ExplorationUserDataModel gets created when creating an exploration.

Reply:
> The ExplorationUserDataModel gets created when creating an exploration.

Yeah, this must is done so that we can add a change list command of creating exploration on the name of the exploration creator and then we can commit this change.

Assume you added a new editor role to an exploration, do we create such a model for that user?

Reply:
Ah I didn't thought about assigning an editor here. Yes that worked. Done!